src/Pages/ChangePassword.cshtml.cs

```diff
@@ -40,21 +47,30 @@ public IActionResult OnGet()
 
         public async Task<IActionResult> OnPost()
         {
-            if (!ModelState.IsValid)
+            try
             {
-                return Page();
-            }
-
-            if (string.Equals(Password, PasswordConfirm))
-            {
-                await UserDatabase.Value.ChangePasswordAsync(User.FindFirst(ClaimTypes.Email).Value, Password);
-                return RedirectToPage("/index");
+                if (!ModelState.IsValid)
+                {
+                    return Page();
+                }
+                if (string.Equals(Password, PasswordConfirm))
+                {
+                    await UserDatabase.Value.ChangePasswordAsync(User.FindFirst(ClaimTypes.Email).Value, Password);
+                    return RedirectToPage("/index");
+                }
+                else
+                {
+                    ErrorMessage = "Passwords did not match!";
+                    return Page();
+                }
             }
-            else
+            catch (Exception ex)
             {
-                ErrorMessage = "Passwords did not match!";
+                _logger.LogInformation(ex.ToString());
+                WasException = true;
```

Comment:
If you mean `wasException` - it stores bool if exception happened and then in the actual csthml (not cshtml.cs) ir allows to load different kind of data using if's (for example not to show date filters etc.)

Reply:
Sorry, I've mixed up

      Because I see that in other pages you use this in the cshtml part, but I can't seem to find it in the html for this page

this comment was meant to be here :D

Reply:
Oh, I actually undestand what you've meant now, yup it got here by mistaking (by copying code). Since I only intended to log exceptions hear regarding accessing database and not redirecting to error page, so bool is not needed here. WIll fix, thanks